### PR TITLE
[Merged by Bors] - chore: move `Init.Data.Quot` to `Logic.Relation`

### PR DESCRIPTION
**Pull request by Parcly-Taxel**

### CODE REVIEW

File: Mathlib/Algebra/RingQuot.lean [hunk 1]
```diff
@@ -73,10 +73,11 @@ theorem Rel.sub_right {R : Type uR} [Ring R] {r : R → R → Prop} ⦃a b c : R
 theorem Rel.smul {r : A → A → Prop} (k : S) ⦃a b : A⦄ (h : Rel r a b) : Rel r (k • a) (k • b) := by
   simp only [Algebra.smul_def, Rel.mul_right h]
 
+open Relation in
 /-- `EqvGen (RingQuot.Rel r)` is a ring congruence. -/
 def ringCon (r : R → R → Prop) : RingCon R where
   r := EqvGen (Rel r)
-  iseqv := EqvGen.is_equivalence _
+  iseqv := Relation.EqvGen.is_equivalence _
```
Comment (semorrison) [inline Mathlib/Algebra/RingQuot.lean:80]: Do you need both of these changes?

File: Mathlib/CategoryTheory/Limits/TypesFiltered.lean [hunk 2]
```diff
@@ -91,13 +92,14 @@ protected theorem rel_equiv : _root_.Equivalence (FilteredColimit.Rel.{v, u} F)
...
 
+open Relation in
 protected theorem rel_eq_eqvGen_quot_rel :
     FilteredColimit.Rel.{v, u} F = EqvGen (Quot.Rel F) := by
   ext ⟨j, x⟩ ⟨j', y⟩
   constructor
   · apply eqvGen_quot_rel_of_rel
   · rw [← (FilteredColimit.rel_equiv F).eqvGen_iff]
-    exact EqvGen.mono (rel_of_quot_rel F)
+    exact Relation.EqvGen.mono (rel_of_quot_rel F)
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Limits/TypesFiltered.lean:102]: Similarly here?

File: Mathlib/Algebra/RingQuot.lean (hunk 1 above)
Comment (Parcly-Taxel) [reply Mathlib/Algebra/RingQuot.lean:80]: The alternative would be to add `Relation.` namespaces everywhere in the declaration.

Comment (Parcly-Taxel) [reply Mathlib/Algebra/RingQuot.lean]: @semorrison I've gone ahead and removed the extraneous `open Relation`s.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**